mitosheet/src/components/taskpanes/Concat/ConcatTaskpane.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from "react";
+import MitoAPI from "../../../api";
+import { ConcatParams, SheetData, UIState } from "../../../types"
+import DropdownButton from "../../elements/DropdownButton";
+import DropdownItem from "../../elements/DropdownItem";
+import Select from "../../elements/Select";
+import SelectAndXIconCard from "../../elements/SelectAndXIconCard";
+import Toggle from "../../elements/Toggle";
+import Col from "../../spacing/Col";
+import Row from "../../spacing/Row";
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+
+
+
+interface ConcatTaskpaneProps {
+    mitoAPI: MitoAPI;
+    sheetDataArray: SheetData[];
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+}
+
+const ConcatTaskpane = (props: ConcatTaskpaneProps): JSX.Element => {
+
+    const [stepID, setStepID] = useState<string | undefined>(undefined);
+    const [concatParams, setConcatParams] = useState<ConcatParams>({
+        join: 'inner',
+        ignore_index: true,
+        sheetIndexes: []
+    })
+    const [updateNumber, setUpdateNumber] = useState(0)
+
+    // Make sure the user cannot select the newly created dataframe
+    const [selectableSheetIndexes] = useState(props.sheetDataArray.map((sd, index) => index));
+
+    useEffect(() => {
+        if (updateNumber === 0) {
+            return;
+        }
+
+        sendConcatEdit()
+    }, [updateNumber])
+
+    // TODO: make it work with undo
```

Comment:
TODO: we need to go and get the params. That's annoying, it would be cool if we had infrastructure for this... might be the time to generate.